server/cluster/cluster_worker.go

```diff
@@ -77,13 +77,17 @@ func (c *RaftCluster) HandleAskSplit(request *pdpb.AskSplitRequest) (*pdpb.AskSp
 	return split, nil
 }
 
-// ValidRequestRegion is used to decide if the region is valid.
-func (c *RaftCluster) ValidRequestRegion(reqRegion *metapb.Region) error {
+// ValidRequestSplitRegion is used to decide if the region is valid.
+func (c *RaftCluster) ValidRequestSplitRegion(reqRegion *metapb.Region) error {
 	startKey := reqRegion.GetStartKey()
 	region := c.GetRegionByKey(startKey)
 	if region == nil {
 		return errors.Errorf("region not found, request region: %v", logutil.RedactStringer(core.RegionToHexMeta(reqRegion)))
 	}
+	// If the region is in joint state, then returns an error.
+	if core.IsInJointState(reqRegion.Peers...) {
+		return errors.Errorf("region is in joint state, request region: %v", logutil.RedactStringer(core.RegionToHexMeta(reqRegion)))
```

Comment:
Why are these errors not defined through the `github.com/pingcap/errors` package?

Reply:
These err need to be returned through GRPC.